datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/UpgradeStep.java

```diff
@@ -31,4 +32,10 @@ default boolean isOptional() {
     return false;
   }
 
+  /**
+   * Returns whether or not to skip the step based on the UpgradeContext
+   */
+  default boolean skip(UpgradeContext context) {
```

Comment:
Question: Why do we need this API if the step can simply check the context and no-op inside it's "execute" method?

Reply:
Because I wanted to mention in the logs that it was skipped. 

Right now it says that DeleteAspectV2TableStep  ran successfully which to me implies that the table was deleted. But it actually wasn't, and was rather just skipped. I think it's good to make this explicit.

Reply:
okay makes sense.. agree its useful to know